packages/protocol/contracts/common/FixidityLib.sol

```diff
@@ -0,0 +1,338 @@
+   * Test newFixedFraction(1,fixed1()) returns 1
+   */
+  function newFixedFraction(
+    uint256 numerator,
+    uint256 denominator
+  )
+    internal
+    pure
+    returns (Fraction memory)
+  {
+    require(numerator <= maxNewFixed());
+    require(denominator <= maxNewFixed());
+    require(denominator != 0);
+    Fraction memory convertedNumerator = newFixed(numerator);
+    Fraction memory convertedDenominator = newFixed(denominator);
+    return divide(convertedNumerator, convertedDenominator);
+  }
+
+  /**
+   * @notice Returns the integer part of a fixed point number.
+   * @dev
+   * Test integer(0) returns 0
+   * Test integer(fixed1()) returns fixed1()
+   * Test integer(newFixed(maxNewFixed())) returns maxNewFixed()*fixed1()
+   */
+  function integer(Fraction memory x) internal pure returns (Fraction memory) {
+    return Fraction((x.value / FIXED1_UINT) * FIXED1_UINT); // Can't overflow
+  }
+
+  /**
+   * @notice Returns the fractional part of a fixed point number.
+   * In the case of a negative number the fractional is also negative.
+   * @dev
+   * Test fractional(0) returns 0
+   * Test fractional(fixed1()) returns 0
+   * Test fractional(fixed1()-1) returns 10^24-1
+   */
+  function fractional(Fraction memory x) internal pure returns (Fraction memory) {
+    return Fraction(x.value - (x.value / FIXED1_UINT) * FIXED1_UINT); // Can't overflow
+  }
+
+  /**
+   * @notice x+y. If any operator is higher than maxFixedAdd() it
+   * might overflow.
+   * @dev
+   * Test add(maxFixedAdd(),maxFixedAdd()) returns maxUint256()-1
+   * Test add(maxFixedAdd()+1,maxFixedAdd()+1) fails
+   * Test add(maxUint256(),maxUint256()) fails
+   */
+  function add(Fraction memory x, Fraction memory y) internal pure returns (Fraction memory) {
+    uint256 z = x.value + y.value;
+    require(z >= x.value);
+    return Fraction(z);
+  }
+
+  /**
+   * @notice x-y.
+   * @dev
+   * Test subtract(6, 10) fails
+   */
+  function subtract(Fraction memory x, Fraction memory y) internal pure returns (Fraction memory) {
+    require(x.value >= y.value);
+    return Fraction(x.value - y.value);
+  }
+
+  /**
+   * @notice x*y. If any of the operators is higher than maxFixedMul() it
+   * might overflow.
+   * @dev
+   * Test multiply(0,0) returns 0
+   * Test multiply(maxFixedMul(),0) returns 0
+   * Test multiply(0,maxFixedMul()) returns 0
+   * Test multiply(fixed1()/mulPrecision(),fixed1()*mulPrecision())
+   * Test multiply(maxFixedMul(),maxFixedMul()) is around maxUint256()
+   * Test multiply(maxFixedMul()+1,maxFixedMul()+1) fails
+   */
+  function multiply(Fraction memory x, Fraction memory y) internal pure returns (Fraction memory) {
+    if (x.value == 0 || y.value == 0) return Fraction(0);
+    if (y.value == FIXED1_UINT) return x;
+    if (x.value == FIXED1_UINT) return y;
+
+    // Separate into integer and fractional parts
+    // x = x1 + x2, y = y1 + y2
+    uint256 x1 = integer(x).value / FIXED1_UINT;
```

Comment:
`integer()` multiplies by FIXED1 and we then divide. If that's our only use of integer we might improve this.